### ISSUE TITLE
[Cards] - Minimize code redundancy

### ISSUE THREAD

**Issue by Luke9389** [TellemBigAlsentya/local-storage-favorites#4]
We have two functions that generate DOM elements for cards (`renderCard` and `renderFavoriteCard`).

While the redundancy in these functions doesn't feel egregious, it could be cool to DRY up the code with the use of helper functions. Let's use this issue to discuss a solution. 

Should we...
 - make helper functions that return DOM elements (and then append them in the parent function)?
 - make helper functions that take an element as a parameter, and append to that element?
 - have only one `renderCard` function and use a parameter to conditionally render DOM? 
 
Which idea do you like, and why?

**Comment by alexhoffman-dev** [comment]
Okay, off the cuff, the last option seems to be the most in line with our current architecture considering we already have a parameter for determining whether the data is being rendered as a favorite or as a character card ie. `grid` vs `faves`. 

With the assistance of some helper functions, I think the updated `renderCard` function could easily consolidate some of the redundant code ie. 
``` js 
    let card = document.createElement('div');
    card.id = characterData.id;
    card.classList.add('characterCard');
    card.addEventListener('click', favorite);
    let cardTitle = document.createElement('div');
    cardTitle.classList.add('characterNames');
    let cardImg = document.createElement('img');

**Comment by Luke9389** [comment]
Ok cool! Sounds good

Feel free to spin up a PR and assign me as a reviewer.

**Comment by alexhoffman-dev** [comment]
@Luke9389  Do you recommend making another new branch as well?

**Comment by Luke9389** [comment]
Yep. Each PR should be on a separate branch. It might be best to merge the current branch you have going first, so let's put this issue on HOLD until [the pagination one is done](https://github.com/TellemBigAlsentya/local-storage-favorites/issues/5).

**Comment by Luke9389** [comment]
🎉 

That PR is done, so we can start this one. Nice work bruv 👾 

So to spin this one up, first go back to your main branch in VS Code and run `git pull`

Then from there, you can make a new branch with `git checkout -b new-branch-name`

Then start your changes for this one, add commit push, and we'll see a PR here.

Also, be sure to link to this issue in the description of the PR so that the PR is easy to find from this page.
